plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -409,6 +409,9 @@ private void testCorruptedTableLocation(String tableName, Path tableLocation, bo
         assertQueryReturnsEmptyResult("SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = CURRENT_SCHEMA AND table_name LIKE 'bad\\_person\\_%' ESCAPE '\\'");
         assertQueryReturnsEmptyResult("SELECT column_name, data_type FROM system.jdbc.columns WHERE table_cat = CURRENT_CATALOG AND table_schem = CURRENT_SCHEMA AND table_name LIKE 'bad\\_person\\_%' ESCAPE '\\'");
 
+        // $history system table returns an empty result for corrupted table
+        assertQueryReturnsEmptyResult("SELECT * FROM \"" + tableName + "$history\"");
```

Comment:
When retrieving  the commit entries `io.trino.plugin.deltalake.DeltaLakeMetadata#getCommitInfoEntries` (or even more low-level `io.trino.plugin.deltalake.transactionlog.checkpoint.TransactionLogTail#loadNewTail(io.trino.filesystem.TrinoFileSystem, java.lang.String, java.util.Optional<java.lang.Long>)` ) we should check whether we have at least one commit entry.

If the table doesn't have any commit entry, something is wrong with it and should be signaled in this kind of query.

If the table has at least one commit entry, but they contain corrupted information (e.g. : protocol entry not found in the transaction log), it would still be a bit controversial to return the history entries of the "corrupted table" successfully.

I find value in doing the extra work by getting the table handle and eventually signaling whether there is an issue.